projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -40,34 +44,53 @@ const ADD_MEDIA_LABEL = __( 'Set Social Image', 'jetpack' );
 const REPLACE_MEDIA_LABEL = __( 'Replace Social Image', 'jetpack' );
 const REMOVE_MEDIA_LABEL = __( 'Remove Social Image', 'jetpack' );
 
-const ALLOWED_MEDIA_TYPES = [ 'image/jpeg', 'image/png' ];
-
 /**
  * Wrapper that handles media-related functionality.
  *
  * @returns {object} The media section.
  */
 export default function MediaSection() {
+	const [ showNotice, setShotNotice ] = useState( false );
 	const { attachedMedia, updateAttachedMedia } = useAttachedMedia();
+	const { connections } = useSocialMediaConnections();
+	const enabledConnections = connections.filter( connection => connection.enabled );
+
+	const mediaValidator = useMemo( () => new MediaValidator( enabledConnections ), [
+		enabledConnections,
+	] );
 
 	const mediaObject = useSelect( select =>
 		select( 'core' ).getMedia( attachedMedia[ 0 ] || null, { context: 'view' } )
 	);
-
 	const { mediaWidth, mediaHeight, mediaSourceUrl } = getMediaDetails( mediaObject );
 
+	useEffect( () => {
+		// Removes selected media if connection change results in invalid image
+		if (
+			mediaObject &&
+			! mediaValidator.isImageValid( mediaObject.media_details?.filesize, mediaObject.mime_type )
+		) {
+			updateAttachedMedia( [] );
+			setShotNotice( true );
+		}
+	}, [ updateAttachedMedia, mediaObject, mediaValidator ] );
+
 	const onRemoveMedia = useCallback( () => updateAttachedMedia( [] ), [ updateAttachedMedia ] );
 	const onUpdateMedia = useCallback(
 		media => {
 			// allowedTypes doesn't properly disallow uploaded media types.
 			// See: https://github.com/WordPress/gutenberg/issues/25130
-			if ( ! ALLOWED_MEDIA_TYPES.includes( media.mime ) ) {
+			// Do not select media if criteria is not met
+			if ( ! mediaValidator.isImageValid( media.filesizeInBytes, media.mime ) ) {
+				updateAttachedMedia( [] );
+				setShotNotice( true );
```

Comment:
I think this might be tricky with the custom hook, as you'd end up calling it inside this callback, which will break the hook rules.

One option might be to always update the media with the new ID, and then allow it to get validated on the next render, but this could lead to a flash of the image.